object-destructuring/exercise-2/exercise.js

```diff
@@ -70,3 +70,19 @@ let hogwarts = [
     occupation: "Teacher",
   },
 ];
+
+// Task 1:
+console.log("Task 1:");
+let GryffindorMembers = hogwarts.filter(({ house }) => house === "Gryffindor");
+GryffindorMembers.forEach(({ firstName, lastName }) => {
+  console.log('${firstName} ${lastName}');
+});
+
+//Task 2
+console.log("Task 2:");
+let teachersWithPets = hogwaters.filter(({ occupation, pet }) => occupation === "Teacher" && pet !== null);
+teachersWithPets.forEach(({ firstName, lastName }) => {
```

Comment:
It is just a matter of coding style, but I prefer to put constants to the left in the comparisons, like this:

`"Teacher" === occupation`

The reason is that if you will try to make a silly mistake like this:

"Teacher" = occupation

JS will not compile and throw an error. Because you can't assign a new value to a constant. While in case of occupation = "Teacher" JS will eagerly accept it as valid.

Reply:
if you want to check for `null` only for `pet`, then you code is ok, but usually people tend to check in general for all falsy values (null/undefined/empty string), in that case a better approach will be:

instead of:

`pet !== null`

use:

`({ occupation, pet }) => "Teacher" === occupation && pet`